core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -1166,9 +1164,7 @@ void writeUInt32Field(int fieldNumber, int value) throws IOException {
       void writeUInt64Field(int fieldNumber, long value) throws IOException {
          int pos = out.getPosition();
          int tag = WireType.makeTag(fieldNumber, WireType.WIRETYPE_VARINT);
-         int tagLen = varIntBytes(tag);
-         int valueLen = varIntBytes(value);
-         out.ensureCapacity(pos + tagLen + valueLen);
+         out.ensureCapacity(pos + 5 + 5);
```

Comment:
I don't think we can assume 5 bytes for the  value here as `Long.MAX_VALUE` requires 9 bytes.

Reply:
Agreed, but it can be up to 10 bytes for a "variable long" afaik.